### PR TITLE
unified kernels registration, test=develop

### PR DESCRIPTION
**Pull request by Shixiaowei02**
### PR types

Function optimization
### PR changes

Others
### Describe

unified kernels registration, test=develop

### CODE REVIEW

File: paddle/infrt/host_context/kernel_registry.h [hunk 1]
```diff
@@ -23,7 +24,7 @@ namespace host_context {
 
 class KernelFrame;
 
-using KernelImplementation = void (*)(KernelFrame *frame);
+using KernelImplementation = std::function<void(KernelFrame *frame)>;
```
Comment (Superjomn) [inline paddle/infrt/host_context/kernel_registry.h:27]:
```c++
auto kernel = [infershape_launcher](KernelFrame* frame) {
  return infershape_launcher(frame);
};

auto kernel = [](KernelFrame* frame) {
  CreateUnintedTensor(frame);
};
```

Hold the InferShapedKernelLauncher instance in the lambda function?

File: paddle/infrt/naive/infershaped/infershaped_kernel_launchers.cc [hunk 2]
```diff
@@ -30,5 +29,19 @@ void RegisterInferShapeLaunchers(InferShapedKernelRegistry* registry) {
...
 }
 
+void RegisterInferShapeLaunchers(host_context::KernelRegistry* registry) {
+  registry->AddKernel(
+      "elementwise_add",
+      std::bind(&KernelLauncherFunc<decltype(&ElementwiseAdd),
+                                    &ElementwiseAdd,
+                                    decltype(&ElementwiseAddInferShape),
+                                    &ElementwiseAddInferShape>,
+                KernelLauncher<decltype(&ElementwiseAdd),
```
Comment (Shixiaowei02) [inline paddle/infrt/naive/infershaped/infershaped_kernel_launchers.cc:39]: put the KernelLauncher object in the function

File: paddle/infrt/host_context/kernel_registry.h (hunk 1 above)
Comment (Shixiaowei02) [reply paddle/infrt/host_context/kernel_registry.h:27]: I fixed it in commit [a759224](https://github.com/PaddlePaddle/Paddle/pull/39583/commits/a759224bdc11e1d1458e83b9c8610cdf385499ca)

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request closed without merging**